### DISCUSSION TITLE
Text rendering: consider removing filter mode settings for text

### DISCUSSION THREAD

**Discussion by morew4rd** [lyte2d/lyte2d#16] [Talk]
There are issues with this mostly related to various artifacts of clamping to pixels two times. (Nearest-neighbor filter first, and "scaling" next)

- Vector representation details get lost, and freetype's anti-alias pixels get the focus at places
- Width changes over longer text lines (See example -- blue is the correct rendering, red is the current scaled with nearest neighbor,)
<img width="902" alt="blue_correct_red_wrong_lyte2d_rendering_scale" src="https://github.com/morew4rd/lyte2d/assets/103900905/ee137af0-69c0-4d1a-a6d8-79715898f3f8">

Need design/decision.

**Comment by morew4rd** [comment]
Marking this as not a bug. But a potential issue for folks who don't know how TTF font rendering works. It's unfortunate but it's kind of hard to make an API without leaky details in this space I feel like.

If anyone sees this and want to talk about your text rendering issues with lyte, ping this one!

**Comment by morew4rd** [comment]
![image](https://github.com/lyte2d/lyte2d/assets/103900905/caa0301c-1b56-452a-a380-9ff47fdf6ad6)

lyte2d code:

```lua
local font = lyte.load_font("assets/DroidSansMono.ttf", 64)

lyte.set_font(font)

function lyte.tick()
    lyte.cls(0, 0, 0, 0)

    lyte.set_color(1,0,0,1)
    lyte.draw_rect(10, 10, 64, 64)
    lyte.draw_rect(10, 10+128, 64, 64)

    lyte.set_color(1,1,1,1)
    lyte.draw_text("Hello, wow 1", 10, 10)
end
```

love2d core:

```lua

local font = love.graphics.newFont("assets/DroidSansMono.ttf", 64)

love.graphics.setFont(font)

function love.draw()
    love.graphics.setColor(1,0, 0, 1)
    love.graphics.rectangle("fill", 10, 10, 64, 64)
    love.graphics.rectangle("fill", 10, 10+128, 64, 64)

    love.graphics.setColor(1,1,1,1)
    love.graphics.print("Hello, wow " .. love.graphics.getDPIScale( ), 10, 10)

end
```

**Reply by lplume** [reply]
![screen](https://github.com/lyte2d/lyte2d/assets/208442/48d0da05-2652-446f-8f11-509f3ad811ea)

**Comment by lplume** [comment]
![screenshot-d245afdd](https://github.com/lyte2d/lyte2d/assets/208442/99a8b931-6d4b-4d38-a713-bdd6820390b8)
love first, lyte is the bottom focused window
not the latest release

**Reply by morew4rd** [reply]
Ah, what size did you load the fonts with?

**Reply by morew4rd** [reply]
Looks like they're about the same size on your machine. I wonder if it's a high-DPI issue on mine?

**Reply by lplume** [reply]
24
